### ISSUE TITLE
Improve Error Reporting: FSI: support for help/quitting/exiting

### ISSUE THREAD

**Issue by jasiozet** [dotnet/fsharp#12616]
It’s not intuitive for new users to remember to add hashtag in front of commands in FSI.

**Current situation:**
```
help;;
error FS0039: The value or constructor 'help' is not defined.
```
```
quit;;
error FS0039: The value or constructor 'quit' is not defined.
```
```
#exit;;
warning FS3353: Invalid directive '#exit 
```
```
exit;;
val it: (int -> 'a)
```

**Why?**
I think these commands are something that any newcomer to FSI might try and be a little discouraged by the results.

I would propose to add this messages to aid users
For example python after quit and exit displays this message:
```
quit
Use exit() or Ctrl-D (i.e. EOF) to exit
```

**Proposed solution:**
This is my general idea (might need to correct the grammar, spacing, etc.)
```
help;;
error FS0039: The value or constructor 'help' is not defined.  
Did you mean #help;; ?
```
```
quit;;
error FS0039: The value or constructor 'quit' is not defined.  
Use #quit;; or Ctrl-D to exit
```
```
#exit;;
warning FS3353: Invalid directive '#exit  
Use #quit;; or Ctrl-D to exit
```
This one is more problematic to give good solution, but here is my attempt:
```
exit;;
val it: (int -> 'a)  
Use exit 0;; #quit;; or Ctrl-D to exit
```

Of course all of them should only happen if you have not defined your own functions named help or quit.

**Additional info**
I just recently discovered F# and I did typed all of these repeatedly until I learned. I tried this experiment on my C# coworkers (type _dotnet fsi_ and try to exit without using help) and most of them struggled until finally being able to exit with _Ctrl+D_ or _Ctrl+Z_ .

**Comment by vzarytovskii** [comment]
@cartermp i remember we were discussing it on standups once or twice. Do you have any related issues in mind which are already logged perhaps?
cc @KathleenDollard

**Comment by cartermp** [comment]
I'm not aware of any related issue. I do know that @KevinRansom once expressed his...happiness...with quitting FSI to some MVPs at a table in 2015. "Hashtag quit, semicolon semicolon. My god, what a mess" or something to that effect.

**Comment by jasiozet** [comment]
Another approach that could be taken (instead or in combination with this) is not enforcing double semicolon to evaluate. Replace it with new line.
This would of course mean that you need _Shift+Enter_ for doing multiline in scope.

**Comment by vzarytovskii** [comment]
> Another approach that could be taken (instead or in combination with this) is not enforcing double semicolon to evaluate. Replace it with new line.
> This would of course mean that you need _Shift+Enter_ for doing multiline in scope.
> But I think that approach needs another discussion and I would bet was already submitted.

Yep, probably better to have it in a separate issue, to not mix those together.
I know that @jonsequitur was experimenting with "semicolon-less" repl using Spectre.Console.

**Comment by jkone27** [comment]
I think neither```;;``` nor multiline should be supported "by default" by interactive on command line, as ofc if you write multiple lines, you can as well open a .fsx file and do your multiline, right? 

what was the original purpose of supporting multiline in fsi terminal prompt?

this makes up for a entry barrier for newcomers I agree
